lightning_examples/finetuning-scheduler/finetuning-scheduler.py

```diff
@@ -0,0 +1,712 @@
+        logger=logger,
+    )
+    trainer.fit(model, datamodule=dm)
+
+
+if AVAIL_GPUS > 0:
+    train()
+else:
+    print("Given the multiple phases of finetuning demonstrated, this notebook is best used with a GPU")
+
+# %% [markdown]
+# ### Running the Baseline and Implicit Finetuning Scenarios
+#
+# Let's now compare our ``nofts_baseline`` and ``fts_implicit`` scenarios with the ``fts_explicit`` one we just ran.
+#
+# We'll need to update our callbacks list, using the core PL ``EarlyStopping`` and ``ModelCheckpoint`` callbacks for the
+# ``nofts_baseline`` (which operate identically to their FTS analogs apart from the recursive training support).
+# For both core PyTorch Lightning and user-registered callbacks, we can define our callbacks using a dictionary as we do
+# with the LightningCLI. This allows us to avoid managing imports and support more complex configuration separated from
+# code.
+#
+# Note that we'll be using identical callback configurations to the ``fts_explicit`` scenario. Keeping [max_depth](https://finetuning-scheduler.readthedocs.io/en/stable/api/finetuning_scheduler.fts.html?highlight=max_depth#finetuning_scheduler.fts.FinetuningScheduler.params.max_depth) for
+# the implicit schedule will limit finetuning to just the last 4 parameters of the model, which is only a small fraction
+# of the parameters you'd want to tune for maximum performance. Since the implicit schedule is quite computationally
+# intensive and most useful for exploring model behavior, leaving [max_depth](https://finetuning-scheduler.readthedocs.io/en/stable/api/finetuning_scheduler.fts.html?highlight=max_depth#finetuning_scheduler.fts.FinetuningScheduler.params.max_depth) 1 allows us to demo implicit mode
+# behavior while keeping the computational cost and runtime of this notebook reasonable. To review how a full implicit
+# mode run compares to the ``nofts_baseline`` and ``fts_explicit`` scenarios, please see the the following
+# [tensorboard experiment summary](https://tensorboard.dev/experiment/n7U8XhrzRbmvVzC4SQSpWw/).
+
+
+# %%
+nofts_callbacks = [EarlyStopping(**earlystopping_kwargs), ModelCheckpoint(**checkpoint_kwargs)]
+fts_implicit_callbacks = [
+    FinetuningScheduler(**fts_kwargs),  # type: ignore # noqa
+    FTSEarlyStopping(**earlystopping_kwargs),  # type: ignore # noqa
+    FTSCheckpoint(**checkpoint_kwargs),  # type: ignore # noqa
+]
+scenario_callbacks = {"nofts_baseline": nofts_callbacks, "fts_implicit": fts_implicit_callbacks}
+
+# %%
+if AVAIL_GPUS > 0:
+    for scenario_name, scenario_callbacks in scenario_callbacks.items():
+        model = RteBoolqModule(**lightning_module_kwargs, experiment_tag=scenario_name)
+        logger = TensorBoardLogger("lightning_logs", name=scenario_name)
+        callbacks = scenario_callbacks
+        print(f"Beginning training the '{scenario_name}' scenario")
+        train()
+else:
+    print("Given the multiple phases of finetuning demonstrated, this notebook is best used with a GPU")
```

Comment:
same here?

Reply:
I've updated the language in my latest commit, def lmk if you prefer a different approach though (e.g. letting training run on CPUs by default)